src/Marten/Linq/SqlGeneration/Filters/CollectionIsEmpty.cs

```diff
@@ -16,7 +16,7 @@ public CollectionIsEmpty(ICollectionMember member)
     {
         _member = member;
         var jsonPath = member.WriteJsonPath();
-        _text = $"d.data @? '$ ? (@.{jsonPath} == null || @.{jsonPath}.size() == 0)'";
+        _text = $"d.data @? '$ ? (@.{member.MemberName} == null || @.{member.MemberName}.size() == 0)'";
```

Comment:
`member.MemberName`, as I understand, will not work with nested collection of inner objects (e.g. Target -> Inner -> Children), but it didn't work with `jsonPath` either.

Another option would be to remove `[*]` at the end of `jsonPath` with Substring, but it seesm it will behave exactly like `.MemberName`, unless I'm missing something.

Reply:
Hey, I'm going to take a longer look at this in the next release with some more test cases -- but still this week!

I've got to get a smaller 7.28.1 out today, but I'll try to roll this into 7.29 later this week.

Thank you for taking this on!

Reply:
Hi @jeremydmiller, did you have chance to look at this yet? Wondering if anything else needed from me

Reply:
I can take it from here. Line 19 above doesn't work though, because the JSON can't be considered to just be teh member name. Some folks will be using Pascal casing like you're assuming, others will use camelCasing that would blow that up. Plus we have to be able to handle deeper nested properties. You kind of fell down the OSS trap of tactically fixing your particular issue, but not seeing the whole picture that I have to worry about. I've got it from here, and thanks for doing what you've done so far as is

Reply:
Thank you, Jeremy! Definitely not insisting the fix is 100% valid, my contribution here is really the unit test, so thanks again for handling it. Nested properties collections didn't work with the previous implementation, so the reasoning was "let's at least fix the basic scenario", didn't really thought about camelCase vs Pascal case, definitely a valid concern.